core/src/main/java/com/linecorp/armeria/client/HttpHeaderSubscriber.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpObject;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.HttpStatusClass;
+
+/**
+ * A {@link Subscriber} that subscribes until it receives {@link HttpHeaders} and then returns a
+ * {@link CompletableFuture} that contains {@link AggregatedHttpMessage}. The message contains
+ * {@link HttpHeaders} and {@code informationalHeaders}. If this class can not subscribe the
+ * {@link HttpHeaders}, it will return {@link AggregatedHttpMessage} with {@link HttpHeaders#EMPTY_HEADERS}.
+ */
+public final class HttpHeaderSubscriber
+        implements Subscriber<HttpObject>, BiConsumer<Void, Throwable> {
```

Comment:
I have some comments for this class but could we move it to `com.linecorp.armeria.internal` until we address them in the future?